opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderTest.java

```diff
@@ -145,16 +148,54 @@ public void testSingleInsert()
         ), resultsIn(mClient,
                 new Assert<>(taskList, new NameData("list1")),
                 new Assert<>(task, new TitleData("task1")),
-                new AssertRelated<>(new InstanceTable(mAuthority), Instances.TASK_ID, task, new AllOf(
-                        new IsNull(Instances.INSTANCE_START),
-                        new EqArg(Instances.INSTANCE_ORIGINAL_TIME, 0),
-                        new IsNull(Instances.INSTANCE_DUE),
-                        new IsNull(Instances.INSTANCE_START_SORTING),
-                        new IsNull(Instances.INSTANCE_DUE_SORTING),
-                        new IsNull(Instances.INSTANCE_DURATION),
-                        new IsNull(Tasks.TZ),
-                        new EqArg(Instances.INSTANCE_STATUS, Instances.INSTANCE_STATUS_NEXT)
-                ))
+                new AssertRelated<>(new InstanceTable(mAuthority), Instances.TASK_ID, task, new InstanceData(Instances.INSTANCE_STATUS_NEXT))
+        ));
+    }
+
+
+    /**
+     * Create 1 local task list and 1 task (via the instances table), check values in TaskLists, TaskList, Instances tables.
+     */
+    @Test
+    public void testInsertSingleInstance()
+    {
+        RowSnapshot<TaskLists> taskList = new VirtualRowSnapshot<>(new LocalTaskListsTable(mAuthority));
+        RowSnapshot<Instances> instance = new VirtualRowSnapshot<>(new InstanceTable(mAuthority));
+
+        assertThat(new Seq<>(
+                new Put<>(taskList, new NameData("list1")),
+                // insert a new task straight into the instances table
+                new Referring<>(taskList, Tasks.LIST_ID, new Put<>(instance, new CharSequenceRowData<>(Tasks.TITLE, "task1")))
+
+        ), resultsIn(mClient,
+                new Assert<>(taskList, new NameData("list1")),
+                new AssertRelated<>(new InstanceTable(mAuthority), Tasks.LIST_ID, taskList,
+                        new Composite<>(
+                                new InstanceData(Instances.INSTANCE_STATUS_NEXT),
+                                new CharSequenceRowData<>(Tasks.TITLE, "task1"))
```

Comment:
Shouldn't this test check the Tasks table as well?
Javadoc says "TasksLists, TaskList, Instances" - should that be "TaskLists, Tasks, Instances"?

Reply:
That "TaskList" is copy'n'pasted from your method ;-)
We definitely should test the task table values as well, but this is out of scope for this story. This story is really just about testing the instance operations. The `InstanceData` class was required to do that properly (the previous solution with `Predicates` was incorrect and could easily miss actual issues).
We'll certainly add more and better tests before we'll release recurrence.

Reply:
> That "TaskList" is copy'n'pasted from your method ;-)
Oh, I see ;) Would you please correct both then?